.github/workflows/codeql-analysis.yml

```diff
@@ -38,11 +38,11 @@ jobs:
 
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v3
+      uses: actions/checkout@d0651293c4a5a52e711f25b41b05b2212f385d28
 
     # Initializes the CodeQL tools for scanning.
     - name: Initialize CodeQL
-      uses: github/codeql-action/init@v2
+      uses: github/codeql-action/init@5a54a995785650c9dbc4c88cc1e1d1b4df4302a8
```

Comment:
All of the codeql-action commit hashes give out a 404. The commit hash for v2 seems to be 3e7e3b32d0fb8283594bb0a76cc60a00918b0969.

https://github.com/github/codeql-action/commit/5a54a995785650c9dbc4c88cc1e1d1b4df4302a8